test/rubygems/test_gem_ext_cargo_builder.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+require_relative 'helper'
+require 'rubygems/ext'
+
+class TestGemExtCargoBuilder < Gem::TestCase
+  def setup
+    @orig_env = ENV.to_hash
+
+    @rust_envs = {
+      'CARGO_HOME' => File.join(@orig_env['HOME'], '.cargo'),
+      'RUSTUP_HOME' => File.join(@orig_env['HOME'], '.rustup'),
+    }
+
+    system(@rust_envs, 'cargo', '-V', out: IO::NULL, err: [:child, :out])
+    pend 'cargo not present' unless $?.success?
+
+    super
+
+    @ext = File.join(@tempdir, 'ext')
+    @dest_path = File.join(@tempdir, 'prefix')
+    @fixture_dir = Pathname.new(File.expand_path('../test_gem_ext_cargo_builder/rust_ruby_example/', __FILE__))
+
+    FileUtils.mkdir_p @dest_path
+    FileUtils.cp_r(@fixture_dir.to_s, @ext)
+  end
+
+  def test_build_staticlib
+    skip_jruby!
+
+    content = @fixture_dir.join('Cargo.toml').read.gsub("cdylib", "staticlib")
+    File.write(File.join(@ext, 'Cargo.toml'), content)
+
+    output = []
+
+    Dir.chdir @ext do
+      ENV.update(@rust_envs)
+      spec = Gem::Specification.new 'rust_ruby_example', '0.1.0'
+      builder = Gem::Ext::CargoBuilder.new(spec)
+      assert_raises(Gem::Ext::CargoBuilder::DylibNotFoundError) do
```

Comment:
Use `assert_raise` instead of `assert_raises`. rubygems and ruby core don't use minitest.

Reply:
This line was reverted to use `assert_raises` again.